exec/java-exec/src/main/java/org/apache/drill/exec/record/AbstractBinaryRecordBatch.java

```diff
@@ -34,9 +35,32 @@
   // state (IterOutcome) of the right input
   protected IterOutcome rightUpstream = IterOutcome.NONE;
 
-  // For now only used by Lateral and Merge Join
   protected RecordBatchMemoryManager batchMemoryManager;
 
+  public final int numInputs = 2;
+  public static final int LEFT_INDEX = 0;
+  public static final int RIGHT_INDEX = 1;
+
+  public enum Metric implements MetricDef {
+    LEFT_INPUT_BATCH_COUNT,
+    LEFT_AVG_INPUT_BATCH_BYTES,
+    LEFT_AVG_INPUT_ROW_BYTES,
+    LEFT_INPUT_RECORD_COUNT,
+    RIGHT_INPUT_BATCH_COUNT,
+    RIGHT_AVG_INPUT_BATCH_BYTES,
+    RIGHT_AVG_INPUT_ROW_BYTES,
+    RIGHT_INPUT_RECORD_COUNT,
+    OUTPUT_BATCH_COUNT,
+    AVG_OUTPUT_BATCH_BYTES,
+    AVG_OUTPUT_ROW_BYTES,
+    OUTPUT_RECORD_COUNT;
```

Comment:
(A repeat of a prior comment on the Hash Join) -- We may be overloading the operator Metrics with low level information that has no value to customers. The Metrics are visible in the web UI.

Reply:
I think they are useful because they give an overall summary for data that is being processed and give insight into what is happening at each operator.